editor/sidebar/post-visibility/index.js

```diff
@@ -56,8 +56,8 @@ class PostVisibility extends Component {
 		const { onUpdateVisibility, onSave } = this.props;
 
 		onUpdateVisibility( 'private' );
+		this.setState( { hasPassword: false } );
```

Comment:
Outside the scope of this pull request, but I don't like how we have to manually manage password state. Seems like it should be inferred by the `password` prop, or at least consolidated to one of the lifecycle methods (e.g. `componentWillUpdate`). I might plan to tackle this separately.